### PR TITLE
Show partial references on 1.3

### PR DESCRIPTION
**Pull request by elliotcm**
If an offer has not yet been accepted and the API version is >= 1.3, show the referee details but `null` out `reference` and `safeguarding_concerns`.

## Things to check

- [x] API release notes have been updated if necessary

I have added a changelog object for this in the new version, but we can't write draft notes for 1.3 while 1.2 is still draft.  There's another ticket here for that:
https://trello.com/c/Bbax3UvE/879-write-the-api-documentation-for-reference-field-nullification

### CODE REVIEW

**Comment by elliotcm** [comment]
[API v1.3: allow reference and safeguarding concerns fields to be nullable](https://trello.com/c/aKb45KSq/822-api-v13-allow-reference-and-safeguarding-concerns-fields-to-be-nullable)

File: app/models/application_reference.rb [hunk 1]
```diff
@@ -41,18 +41,19 @@ def self.requested_or_provided
...
+  TEMPORARY_FAILURE_FEEDBACK_STATUS = %i[email_bounced].freeze
 
   def self.failed
-    where(feedback_status: FAILED_FEEDBACK_STATUS)
+    where(feedback_status: PERMANENT_FAILURE_FEEDBACK_STATUS)
   end
 
   def self.not_failed
-    where.not(feedback_status: FAILED_FEEDBACK_STATUS)
+    failed.invert_where
```
Comment (carlosmartinez) [inline app/models/application_reference.rb:52]: dear lord, yes

File: spec/presenters/vendor_api/v1.0/application_presenter_spec.rb [hunk 2]
```diff
@@ -82,19 +82,7 @@
...
-    end
-
-    it 'includes the value of include_incomplete_references in the cache key' do
-      allow(FeatureFlag).to receive(:feature_statuses).and_return({})
-      allow(Rails.cache).to receive(:fetch)
-
-      described_class.new(version, application_choice, include_incomplete_references: true).serialized_json
-
-      hashed_hash = { incomplete_references: true }.hash
+      hashed_hash = {}.hash
```
Comment (carlosmartinez) [inline spec/presenters/vendor_api/v1.0/application_presenter_spec.rb:85]: so much hash

File: spec/presenters/vendor_api/v1.3/application_presenter_spec.rb [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  describe '#references' do
+    let(:reference) do
+      build(:reference, reference_status)
+    end
+
+    let(:reference_status) { :feedback_provided }
+
+    before do
+      application_form.application_references.destroy_all
```
Comment (carlosmartinez) [inline spec/presenters/vendor_api/v1.3/application_presenter_spec.rb:17]: might be useful to have `:without_references` as an application choice factory trait

**Review by carlosmartinez — APPROVED** [review]
great

**Pull request merged**